evalml/data_checks/invalid_target_data_check.py

```diff
@@ -244,17 +244,24 @@ def _check_target_has_nan(self, y, results):
                 ),
                 results,
             )
-            impute_strategy = (
-                "mean" if is_regression(self.problem_type) else "most_frequent"
-            )
+
             results["actions"]["action_list"].append(
-                DataCheckAction(
+                DataCheckActionOption(
                     DataCheckActionCode.IMPUTE_COL,
                     data_check_name=self.name,
-                    metadata={
-                        "is_target": True,
-                        "impute_strategy": impute_strategy,
+                    parameters={
+                        "impute_strategy": {
```

Comment:
Yeah, I think during the design phase we had moved away from that since it introduced more complexity but as I wrote this out, I think this is a great suggestion. Will update, thank you 🙏